app/controllers/ems_common.rb

```diff
@@ -566,6 +567,37 @@ def recheck_authentication(id = nil)
     @record.authentication_check_types_queue(@record.authentication_for_summary.pluck(:authtype), :save => true)
   end
 
+  def check_compliance(model)
+    # assert_privileges("#{model.name.underscore}_check_compliance")
+    showlist = @lastaction == "show_list"
+    ids = showlist ? find_checked_items : find_current_item(model)
+
+    if ids.empty?
+      add_flash(_("No %{model} were selected for %{task}") % {:model => ui_lookup(:models => model),
+                                                              :task  => "Compliance Check"}, :error)
+    else
+      process_check_compliance(model, ids)
```

Comment:
@alongoldboim i think you don't need to add a new method 'process_check_compliance' in this PR, i think you should be able to call existing method 'process_emss' https://github.com/ManageIQ/manageiq/blob/master/app/controllers/ems_common.rb#L987 instead.